demo/admin/src/products/categories/ProductCategoriesGrid.tsx

```diff
@@ -0,0 +1,200 @@
+import * as React from "react";
+import { useIntl } from "react-intl";
+
+import {
+    GQLCreateProductCategoryMutation,
+    GQLCreateProductCategoryMutationVariables,
+    GQLDeleteProductCategoryMutation,
+    GQLDeleteProductCategoryMutationVariables,
+    GQLProductCategoriesGridQuery,
+    GQLProductCategoriesGridQueryVariables,
+    GQLProductCategoryGridFutureFragment,
+    GQLUpdateProductCategoryPositionMutation,
+    GQLUpdateProductCategoryPositionMutationVariables,
+} from "./ProductCategoriesGrid.generated";
+
+const productCategoriesFragment = gql`
+    fragment ProductCategoryGridFuture on ProductCategory {
+        id
+        title
+        slug
+        position
+    }
+`;
+
+const productCategoriesQuery = gql`
+    query ProductCategoriesGrid($offset: Int!, $limit: Int!, $sort: [ProductCategorySort!]) {
+        productCategories(offset: $offset, limit: $limit, sort: $sort) {
+            nodes {
+                ...ProductCategoryGridFuture
+            }
+            totalCount
+        }
+    }
+    ${productCategoriesFragment}
+`;
+
+const deleteProductCategoryMutation = gql`
+    mutation DeleteProductCategory($id: ID!) {
+        deleteProductCategory(id: $id)
+    }
+`;
+
+const createProductCategoryMutation = gql`
+    mutation CreateProductCategory($input: ProductCategoryInput!) {
+        createProductCategory(input: $input) {
+            id
+        }
+    }
+`;
+
+const updateProductCategoryPositionMutation = gql`
+    mutation UpdateProductCategoryPosition($id: ID!, $input: ProductCategoryUpdateInput!) {
+        updateProductCategory(id: $id, input: $input) {
+            id
+            updatedAt
+            position
+        }
+    }
+`;
+
+function ProductCategoriesGridToolbar({ toolbarAction }: { toolbarAction?: React.ReactNode }) {
+    return (
+        <DataGridToolbar>
+            <ToolbarActions>{toolbarAction}</ToolbarActions>
+        </DataGridToolbar>
+    );
+}
+
+type Props = {
+    toolbarAction?: React.ReactNode;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    rowAction?: (params: GridRenderCellParams<any, GQLProductCategoryGridFutureFragment, any>) => React.ReactNode;
+    actionsColumnWidth?: number;
+};
+
+export function ProductCategoriesGrid({ toolbarAction, rowAction, actionsColumnWidth = 52 }: Props): React.ReactElement {
+    const client = useApolloClient();
+    const intl = useIntl();
+    const dataGridProps = { ...useDataGridRemote(), ...usePersistentColumnState("ProductCategoriesGrid") };
+
+    const handleRowOrderChange = async ({ row: { id }, targetIndex }: GridRowOrderChangeParams) => {
+        await client.mutate<GQLUpdateProductCategoryPositionMutation, GQLUpdateProductCategoryPositionMutationVariables>({
+            mutation: updateProductCategoryPositionMutation,
+            variables: { id, input: { position: targetIndex + 1 } },
+            awaitRefetchQueries: true,
+            refetchQueries: [productCategoriesQuery],
```

Comment:
Why is this necessary?

Reply:
We'll probably still need it, because the items would jump back to the original position based on the Apollo cache I suppose.